### DISCUSSION TITLE
Kong DP deployment model

### DISCUSSION THREAD

**Discussion by shettyh** [Kong/kong#11644] [Help] [answered]
Hi, 

I have questions related to Kong DP deployment modes, 

- Major concerns are around when the data size is huge ( like lot of credentials and consumers ), How much time DP takes to start up ? What is generally recommended for such cases ? Is having a EBS volume attached to each pod is recommended ? 

- If the config/data is not yet fully loaded, data plane still starts to serve traffic ?

Could not get these details from docs or from code. Would be helpful if someone can give these details

Thanks

**Comment by StarlightIbuki** [comment, marked as answer]
1. It depends. Many things affect the time: the scale of your config, the hardware, the load, etc. An extreme config results in a dozen minutes of delay to apply, while a usual config takes seconds. We are working on improving performance on large configs set and it will land in future versions of Kong;
2. DP won't serve with partially loaded config. However, DP continues to serve traffic with the old configuration while loading the new config.
